Code/Framework/AzFramework/AzFramework/Physics/DebugDraw/CharacterPhysicsDebugDraw.cpp

```diff
@@ -44,9 +44,23 @@ namespace Physics
         for (size_t colliderIndex = 0; colliderIndex < colliders.size(); colliderIndex++)
         {
             const auto& collider = colliders[colliderIndex];
-            const AZ::Color colliderColor = ((1 << colliderIndex) & invalidColliderBitArray)
-                ? colorSettings.m_errorColor
-                : (nodeDebugDrawData.m_selected ? colorSettings.m_selectedColor : colorSettings.m_defaultColor);
+            AZ::Color colliderColor;
+            if (nodeDebugDrawData.m_hovered)
+            {
+                colliderColor = colorSettings.m_hoveredColor;
+            }
+            else if (nodeDebugDrawData.m_selected)
+            {
+                colliderColor = colorSettings.m_selectedColor;
+            }
+            else if ((1 << colliderIndex) & invalidColliderBitArray)
+            {
+                colliderColor = colorSettings.m_errorColor;
+            }
+            else
+            {
+                colliderColor = colorSettings.m_defaultColor;
+            }
```

Comment:
suggestion (feel free to ignore): You could move this to an immediately invoked lambda so `colliderColor` can be `const`. This nicely separates initialization from other uses.

```
const AZ::Color colliderColor = [] {
    if (nodeDebugDrawData.m_hovered)
    {
        return colorSettings.m_hoveredColor;
    }
    else if (nodeDebugDrawData.m_selected)
    {
        return olorSettings.m_selectedColor;
    }
    else if ((1 << colliderIndex) & invalidColliderBitArray)
    {
        return colorSettings.m_errorColor;
    }
    return colorSettings.m_defaultColor;
}();
```

Reply:
I tried that out, but I had to capture so many things in the lambda that it felt a bit clunky.